pyproject.toml

```
@@ -24,6 +24,7 @@ ordered-set = ">=4.1.0"
 typer = "^0.7.0"
 typer-cli = "^0.0.13"
 loguru = "*"
+sssom = ">=0.3.31"
```

```suggestion
sssom = ">=0.3.40"
```

Some _very_ important changes about the mapping justification IRIs came in in 36 or so.

---

Unfortunately, upgrading to 0.3.40 breaks the SSSOM test, and I'm not sure why. 
Reverting to 0.3.31 for the time being.

---

What's the error message? This change is pretty important so you are not hit by surprise in the next pr. most likely the curie map in you SSSOM test has the wrong value for the semapv prefix.

---

TLDR:
```
=================================================================================== short test summary info ====================================================================================
FAILED tests/unit/test_sssom_mapping.py::test_basic_mapping - KeyError: "['predicate_modifier'] not in index"
FAILED tests/unit/test_sssom_mapping.py::test_merge_and_filter - KeyError: "['predicate_modifier'] not in index"
FAILED tests/unit/test_sssom_mapping.py::test_exact_match_is_bidirectional - KeyError: "['predicate_modifier'] not in index"
===================================================================== 3 failed, 107 passed, 1 warning in 71.45s (0:01:11) ======================================================================
```

For full output (it's quite long) I would check out [one of the failed github actions](https://github.com/monarch-initiative/koza/actions/runs/6076500179/job/16484614849)

---

@hrshdhgd can you look into this and advice?

---

While using `filter_prefixes`, `features` should be provided as name of columns to be grouped by. The default is `KEY_FEATURES = [SUBJECT_ID, PREDICATE_ID, OBJECT_ID, PREDICATE_MODIFIER]` in `sssom`. 

`features` is not provided in the test here and hence the default is assumed. `predicate_modifier` is not a column in the test data and hence the error. simple fix:

In `source_config.py`:
```
new_msdf = filter_prefixes(
            df=new_msdf.df,
            filter_prefixes=filters,
            require_all_prefixes=False,
            features=new_msdf.df.columns
        )
```

I took the liberty of updating the code and updating `sssom` to the latest version. Apologize if it upsets anything.

---

Thank you @hrshdhgd